dev/generate_test_trajectory.ts

```diff
@@ -1,144 +1,450 @@
-          for (const waypoints of track.segments) {
-            for (const { lat, lon, time } of waypoints) {
-              coordinates.push([lat, lon])
-              timestamps.push(time)
-            }
-          }
-        }
-        const meta = { id, placename, type: TrajectoryType.EXAMPLE }
-        const data = { coordinates, timestamps }
-        resolve(new Trajectory(meta, data))
-      })
-    })
-  }
 }
 
 function createCluster(
   trajectory: Trajectory,
   numberPoints: number = 30,
-  radius: number = 15
+  minRadius: number = 3,
+  maxRadius: number = 30
 ): Trajectory {
-  const seed = trajectory.coordinates[trajectory.coordinates.length - 1]
+  const resultTrajectory = trajectory.getCopy()
+  const seed =
+    resultTrajectory.coordinates[resultTrajectory.coordinates.length - 1]
   for (var i: number = 0; i < numberPoints; i++) {
-    const rand = randomGeo(seed[0], seed[1], radius)
-    trajectory.coordinates.push([rand.latitude, rand.longitude])
-    trajectory.timestamps.push(null)
+    const rand = randomGeo(seed[0], seed[1], minRadius, maxRadius)
+    resultTrajectory.coordinates.push([rand.latitude, rand.longitude])
+    resultTrajectory.timestamps.push(null)
+  }
+  return resultTrajectory
+}
+
+function addTimestampsForTrajectory(
+  first: Date,
+  last: Date,
+  trajectory: Trajectory
+): Trajectory {
+  const duration = last.getTime() - first.getTime()
+  const numberTimestamps = trajectory.coordinates.length
+  const stepLength = duration / numberTimestamps
+  const resultTrajectory = trajectory.getCopy()
+  for (let i = 0; i < numberTimestamps; i++) {
+    resultTrajectory.timestamps[i] = new Date(first.getTime() + i * stepLength)
   }
-  return trajectory
+  return resultTrajectory
+}
+
+/**
+ * Interpolates and adds additional coordinates in between given set of coordinates.
+ * @param trajectory which holds coordinates
+ * @param numberOfInsertsPerCoordinate number of interpolated points between each pair of subsequent coordinates
+ * @returns interpolated trajectory
+ */
+function interpolateCoordinates(
+  trajectory: Trajectory,
+  numberOfInsertsPerCoordinate: number = 1
+) {
+  if (numberOfInsertsPerCoordinate < 1) {
+    return trajectory
+  }
+  const resultTrajectory = trajectory.getCopy()
+  const numberCoordinates = trajectory.coordinates.length
+  for (let index = 0, indexShift = 0; index < numberCoordinates - 1; index++) {
+    const [firstLat, firstLng] = trajectory.coordinates[index]
+    const [secondLat, secondLng] = trajectory.coordinates[index + 1]
+    const latDiff = firstLat - secondLat
+    const lngDiff = firstLng - secondLng
+    for (
+      let insertIndex = 1;
+      insertIndex <= numberOfInsertsPerCoordinate;
+      insertIndex++, indexShift++
+    ) {
+      const latSeed =
+        firstLat + latDiff * (insertIndex / numberOfInsertsPerCoordinate)
+      const lngSeed =
+        firstLng + lngDiff * (insertIndex / numberOfInsertsPerCoordinate)
```

Comment:
Here, we treat latitude and longitude like Cartesian coordinates on a flat plane. In fact, lat and lon are coordinates on an ellipsoid and thus we'd have to interpolate between two coordinates on an arc and not a straight line. But since we are not treating long distances, I think we can ignore that for now. ;-)

Reply:
Agreed - I wanted to keep it simple for now.